test/clj/rems/api/test_catalogue.clj

```diff
@@ -19,15 +19,47 @@
       (is (some #(str/starts-with? (:resid %) "urn:") data)))))
 
 (deftest catalogue-api-security-test
-  (testing "listing without authentication"
-    (let [response (-> (request :get (str "/api/catalogue"))
-                       handler)
-          body (read-body response)]
-      (is (response-is-unauthorized? response))
-      (is (= "unauthorized" body))))
-  (testing "listing with wrong API-Key"
-    (is (= "unauthorized"
-           (-> (request :get (str "/api/catalogue"))
-               (assoc-in [:headers "x-rems-api-key"] "invalid-api-key")
-               handler
-               (read-body))))))
+  (with-redefs [rems.config/env (assoc rems.config/env :catalogue-is-public true)]
+    (testing "catalogue-is-public true with no authentification"
+      (let [response (-> (request :get "/api/catalogue")
+                         handler)]
+        (is (response-is-ok? response))))
+    (let [api-key "42"]
+      (testing "catalogue-is-public true with authentification by api key"
+        (let [response (-> (request :get "/api/catalogue")
+                           (authenticate api-key "handler")
+                           handler)]
+          (is (response-is-ok? response)))))
+    (let [api-key "42"
+          user-id "alice"]
+      (testing "catalogue-is-public true with authentification by api key and username"
```

Comment:
You could move `catalogue-is-public` to a new `testing` block above where the `with-redefs` is so that you don't need to duplicate it everywhere.

Reply:
I am sorry, I didnt get it

Reply:
This string `"catalogue-is-public true ` is repeated in every `testing` block here. This structure would be better with less repetition?

```
(testing "catalogue-is-public"
  ...
  (testing "authenticaiton by api key and username"
    ...)
  (testing "aithentication by api key"
    ...)
  ...)

Reply:
I agree, I ll do the change